hyperdome_client/hyperdome_client.py

```diff
@@ -486,15 +434,42 @@ def copy_hidservauth(self):
         self.system_tray.showMessage(strings._('gui_copied_hidservauth_title'),
                                      strings._('gui_copied_hidservauth'))
 
-    def closeEvent(self, e):
+    def disconnect_chat(self):
+        self.start_chat_button.setEnabled(False)
+        self.timer.stop()
+        self.worker.clear()
+        self.get_messages_task.signals.success.disconnect(
+            self.on_history_added)
+        if self.get_messages_task is not None:
+            del self.get_messages_task
```

Comment:
This object is setup in such a way that Qt is actually hiding it from python's garbage collector. this is done with `self.get_messages_task.setAutoDelete(False)` which tells Qt that you want to handle the life-cycle of this thread. Otherwise it is likely that a server timeout will cause the thread to die and stop retrieving messages.

Reply:
Interesting :o